src/components/UserActions/UserActivity.ts

```diff
@@ -103,126 +93,167 @@ export class UserActivity extends Component {
+            return null;
+        }
+    }
 
-        const cause = (isCustom && action.raw.event_value) || (isSearch && action.raw.cause) || '';
+    private isPartOfTheSameSession = (action: UserAction, previousDateTime: Date): boolean => {
+        return Math.abs(action.timestamp.valueOf() - previousDateTime.valueOf()) / DATE_TO_SECONDS / DATE_TO_MINUTES < MAX_MINUTES_IN_SESSION;
+    };
 
-        const useInclude = this.options.unfoldInclude && this.options.unfoldInclude.length > 0;
+    private splitActionsBySessions(actions: UserAction[]): UserActionSession[] {
+        if (actions.length === 0) {
+            return [];
+        }
+        let splitSessions: UserActionSession[] = [];
+        splitSessions.push(new UserActionSession(actions[0].timestamp, []));
+        let previousDateTime = actions[0]?.timestamp;
+        let currentSession: UserActionSession = splitSessions[0];
+        actions.forEach((action) => {
+            if (this.isPartOfTheSameSession(action, previousDateTime)) {
+                currentSession.actions.push(action);
+            } else {
+                splitSessions.push(new UserActionSession(action.timestamp, [action]));
+                currentSession = splitSessions[splitSessions.length - 1];
+            }
+            previousDateTime = action.timestamp;
+        });
+        return splitSessions;
+    }
 
-        const isExcluded = (isSearch || isCustom) && this.options.unfoldExclude.indexOf(cause) !== -1;
-        const isIncluded = (isSearch || isCustom) && this.options.unfoldInclude.indexOf(cause) !== -1;
+    private buildSessionsToDisplay() {
+        if (this.options.ticketCreationDateTime instanceof Date) {
+            ({ caseSubmitSessionIndex: this.caseSubmitSessionIndex, caseSubmitSession: this.caseSubmitSession } = this.findCaseSubmitSession());
+            if (this.caseSubmitSessionIndex !== -1) {
+                this.sessionsToDisplay = this.findSurroundingSessions();
+                this.caseSubmitSession.expanded = true;
+
+                const insertTicketCreatedIndex = this.caseSubmitSession.actions.findIndex(
+                    (action) => action.timestamp <= this.options.ticketCreationDateTime
+                );
+                this.caseSubmitSession.actions.splice(insertTicketCreatedIndex, 0, this.buildTicketCreatedAction());
+                return;
+            } else {
+                console.warn(`Could not find a user action session corresponding to this date: ${this.options.ticketCreationDateTime}.`);
+            }
+        }
+        this.sessionsToDisplay = this.sessions.slice(0, 5);
+        this.sessionsToDisplay[0].expanded = true;
+    }
+
+    private buildTicketCreatedAction(): UserAction {
+        return new UserAction(UserActionType.TicketCreated, this.options.ticketCreationDateTime, {});
+    }
+
+    private findCaseSubmitSession(): { caseSubmitSessionIndex: number; caseSubmitSession: UserActionSession } {
```

Comment:
I'd rather keep the original order since for use-cases we want to add the ticket creation event to a session if the "end" is closer, I'd rather work with `[0]` than `[length-1]`

I will try to refactor to clarify.